src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -0,0 +1,149 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Logging;
+    using ObjectBuilder;
+    using Transport;
+
+    class ReceiveComponent
+    {
+        public ReceiveComponent(ReceiveConfiguration configuration, TransportReceiveInfrastructure receiveInfrastructure, QueueBindings queueBindings)
+        {
+            this.configuration = configuration;
+            this.receiveInfrastructure = receiveInfrastructure;
+            this.queueBindings = queueBindings;
+        }
+
+        public async Task Initialize(IPipelineExecutor mainPipelineExecutor, IEventAggregator eventAggregator, IBuilder builder, CriticalError criticalError, string errorQueue)
+        {
+            if (!configuration.IsEnabled)
+            {
+                return;
+            }
+
+            if (configuration.PurgeOnStartup)
+            {
+                Logger.Warn("All queues owned by the endpoint will be purged on startup.");
+            }
+
+            AddReceivers(mainPipelineExecutor, eventAggregator, builder, criticalError, errorQueue);
+
+            foreach (var receiver in receivers)
+            {
+                try
+                {
+                    await receiver.Init().ConfigureAwait(false);
```

Comment:
Initializing here means we are doing it much earlier than we used to, and this can break transports. For example, we're now calling Init before queues have had a chance to be created, so if `PurgeOnStartup` is true, queues will be attempted to be purged in RabbitMQ before the queue has been created: https://github.com/Particular/NServiceBus.RabbitMQ/blob/develop/src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs#L67

This change also means we're calling init before we've called the receive PreStartupCheck. That could potentially cause a problem as well, though it doesn't specifically for RabbitMQ.

Reply:
Good catch, I've move the calls back to the original place.

Observation: It feels strange that we call "PreStartUpChecks" before we call .Init?